### PR TITLE
adding backup script

### PR DESCRIPTION
**Pull request by colins44**
This allows us to back up redis from SFMC using the command given within package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "build": "rm -rf dist && tsc",
     "lint": "eslint src --ext .js,.ts",
     "lint-fix": "yarn run lint --fix",
+    "backUpFromSFMC": "node -e 'require(\"./dist/scripts/sfmcBackUp\").backUpFromSFMC()'",
```
Comment (joshuacullenlux) [inline package.json:12]: What do you think about using the : format?

```
backup:sfmc
```

File: src/controllers/popularLocationController.ts [hunk 2]
```diff
@@ -53,7 +53,7 @@ export async function remove(
 ): Promise<Response | void> {
   await deletePopularLocation(req.params.tag);
   res.status(204);
-  return res.end()
+  return res.end();
```
Comment (joshuacullenlux) [inline src/controllers/popularLocationController.ts:56]: Does this repo have the magic standard/eslint?

File: src/scripts/sfmcBackUp.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import redis from "../lib/redis";
+import { getKey } from "../lib/redisFunctions";
+import {
+  buildObjectsFromSoapResponse,
+  getRequest
+} from "../services/marketingCloud";
+
+export async function getUserAlerts(requestId?: string): Promise<any> {
+  const dataExtenstionExternalKey: string =
+    process.env.USER_ALERT_DE_EXTERNAL_KEY || "";
```
Comment (joshuacullenlux) [inline src/scripts/sfmcBackUp.ts:10]: What happens if this is the empty string? Will it just return empty?

File: src/scripts/sfmcBackUp.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  );
+  return response;
+}
+
+export async function saveUserAlerts(requestId?: string): Promise<void> {
+  const response = await getUserAlerts(requestId);
+  const userAlerts = buildObjectsFromSoapResponse(response);
+  const pipeline = redis.pipeline();
+  userAlerts.forEach(alt => {
+    const key = getKey(alt.heroku_id, "destinationAlerts");
```
Comment (joshuacullenlux) [inline src/scripts/sfmcBackUp.ts:61]: Have you thought about putting in some redis models?

That way you coould abstract away the internals for doing the save and just call something like

```
DestinationAlertsModel.save({userAlerts})
```

File: src/scripts/sfmcBackUp.ts (hunk 3 above)
Comment (colins44) [reply src/scripts/sfmcBackUp.ts:10]: App crashes, this is more for typescript errors

File: src/scripts/sfmcBackUp.ts (hunk 4 above)
Comment (colins44) [reply src/scripts/sfmcBackUp.ts:61]: not sure that we need another layer of abstraction for saving a hash to redis within a script

File: src/controllers/popularLocationController.ts (hunk 2 above)
Comment (colins44) [reply src/controllers/popularLocationController.ts:56]: Very close to standard, there might of been a tweak or two. Anything you dont like?

File: package.json (hunk 1 above)
Comment (colins44) [reply package.json]: sounds good, changing now

File: src/controllers/popularLocationController.ts (hunk 2 above)
Comment (joshuacullenlux) [reply src/controllers/popularLocationController.ts:56]: Nah all g :+1: Was wondering why it didn't pick this up. 

Maybe a full --fix on the repo?

File: src/scripts/sfmcBackUp.ts (hunk 4 above)
Comment (joshuacullenlux) [reply src/scripts/sfmcBackUp.ts:61]: Yeah you're right but then eventually you might need to do the same elsewhere. I always find it nice to encapsulate the model stuff. Saves future me a lot of hassle.

**Pull request merged**